internal/resolve/catalog.go

```diff
@@ -95,31 +96,23 @@ func (r *CatalogResolver) Resolve(ctx context.Context, ext *ocv1alpha1.ClusterEx
 			return compare.ByVersion(a, b)
 		})
 
-		thisBundle := packageFBC.Bundles[0]
-		if resolvedBundle != nil {
-			// Cases where we stick with `resolvedBundle`:
-			//  1. If `thisBundle` is deprecated and `resolvedBundle` is not
-			//  2. If `thisBundle` and `resolvedBundle` have the same deprecation status AND `resolvedBundle` is a higher version
-			if isDeprecated(thisBundle, thisDeprecation) && !isDeprecated(*resolvedBundle, resolvedDeprecation) {
-				return nil
-			}
-			if compare.ByVersion(*resolvedBundle, thisBundle) < 0 {
-				return nil
-			}
-		}
-		resolvedBundle = &thisBundle
+		resolvedBundles = append(resolvedBundles, &packageFBC.Bundles[0])
+		matchedCatalogs = append(matchedCatalogs, cat.GetName())
 		resolvedDeprecation = thisDeprecation
```

Comment:
The updated code does not use deprecation to disambiguate, and I think it needs to. If we have the case where two bundles are found, but one of them is deprecated, we should not error out and should return the non-deprecated bundle.

Specifically, if `resolvedBundles` contains a deprecated bundle, and the new bundle is non-deprecated, we should ignore the deprecated bundles in `resolvedBundles`, and continue to add only non-deprecated bundles. Something like:

if list empty: add bundle to list, and update `resolvedDeprecation`
if list not empty:
  if `thisBundle` is not deprecated and list is deprecated: clear out list and add `thisBundle` to list and update `resolvedDeprecation`
  if `thisBundle` is deprecated and list is not deprecated: ignore new deprecated `thisBundle`
  otherwise (both new bundle and list share deprecation status): add `thisBundle` to list
  
What this does is make a homogeneous list of either all deprecated or all non-deprecated bundles. If we _then_ have multiple bundles, then it's an error.

Also note that `resolvedDeprecation` is always the last deprecation status, regardless of any other bundle on the list.

Reply:
I may have simply remembered this wrong; I was under the impression that we'd give an ambiguous resolution error in this case as well. If that wasn't the agreement I'll put this back as a deciding factor.

Reply:
Deprecated was definitely kept; it was just version being the tie breaker. You have a good approach, because you have the list of found operations.